File: src/main/java/io/github/pixee/maven/operator/VersionDefinition.java
package io.github.pixee.maven.operator;

public class VersionDefinition {
    private Kind kind;
    private String value;

    public VersionDefinition(Kind kind, String value) {
        this.kind = kind;
        this.value = value;
    }

    public void setKind(Kind kind) {
        this.kind = kind;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public Kind getKind() {
        return kind;
    }

    public String getValue() {
        return value;
    }
}

File: src/main/java/io/github/pixee/maven/operator/VersionDefinition.kt
package io.github.pixee.maven.operator

import io.github.pixee.maven.operator.java.KindJ

/**
 * Represents a tuple (kind / version string) applicable from a pom.xml file
 *
 * For Internal Consumption (thus Internal)
 */
internal data class VersionDefinition(
    val kind: KindJ,
    val value: String,
)